linalg/solvers.cpp

```diff
@@ -103,6 +104,49 @@ void IterativeSolver::SetOperator(const Operator &op)
 }
 
 
+OperatorJacobiSmoother::OperatorJacobiSmoother(const Vector &d,
+                                               const Array<int> &ess_tdofs,
+                                               const double dmpng)
+   :
+   Solver(d.Size()),
+   N(d.Size()),
+   dinv(N),
+   diag(d),
+   damping(dmpng),
+   ess_tdof_list(ess_tdofs),
+   residual(N) { Setup(); }
+
+void OperatorJacobiSmoother::Setup()
+{
+   residual.UseDevice(true);
+   const double delta = damping;
+   auto D = diag.Read();
+   auto DI = dinv.Write();
+   MFEM_FORALL(i, N, DI[i] = delta / D[i]; );
+   auto I = ess_tdof_list.Read();
+   MFEM_FORALL(i, ess_tdof_list.Size(), DI[I[i]] = delta; );
```

Comment:
This line too:
`linalg/solvers.cpp(127): error: The enclosing parent function ("Setup") for an extended __device__ lambda cannot have private or protected access within its class`

Reply:
> I would expect `Setup` to be private since it's only used in the constructor, do you foresee a user calling this function?

By moving `Setup` to `private`, yes it led to new issues.